apis/placement/v1beta1/clusterresourceplacement_types.go

```diff
@@ -148,6 +148,10 @@ type PlacementSpec struct {
 	// +kubebuilder:default=10
 	// +kubebuilder:validation:Optional
 	RevisionHistoryLimit *int32 `json:"revisionHistoryLimit,omitempty"`
+
+	// DeleteOptions configures the deletion behavior when the ClusterResourcePlacement is deleted.
+	// +kubebuilder:validation:Optional
+	DeleteOptions *DeleteOptions `json:"deleteOptions,omitempty"`
```

Comment:
@ryanzhang-oss @Arvindthiru 

A few open questions:
1. Shall we make DeleteOptions side-by-side with RolloutStrategy or make it a child of RolloutStrategy? I am not strong on this. It is somewhat unrelated to roll out. It is about what happens if the CRP is deleted, rather than what happens if the selected resources change or the placement policy changes.

2. I think our current behavior is "abandon" meaning keeping the placed resources when CRP is deleted. Is that true?

Reply:
After with @ryanzhang-oss, we are inclined to place DeleteStrategy inside RolloutStrategy to mirror how ApplyStraegy is used. RolloutStrategy can be changed in the future to just be Strategy in the API since it's just defined as strategy in the CRD https://github.com/Azure/fleet/blob/main/config/crd/bases/placement.kubernetes-fleet.io_clusterresourceplacements.yaml#L621